crates/ere-cli/src/main.rs

```diff
@@ -0,0 +1,259 @@
+
+    #[cfg(feature = "sp1")]
+    let program = compile!(ere_sp1::RV32_IM_SUCCINCT_ZKVM_ELF);
+
+    #[cfg(feature = "zisk")]
+    let program = compile!(ere_zisk::RV64_IMA_ZISK_ZKVM_ELF);
+
+    fs::write(&program_path, program)
+        .with_context(|| format!("Failed to write program to {}", program_path.display()))
+}
+
+fn prove(
+    program_path: PathBuf,
+    resource: ProverResourceType,
+    input_path: PathBuf,
+    proof_path: PathBuf,
+    report_path: PathBuf,
+) -> Result<(), Error> {
+    let zkvm = construct_zkvm(program_path, resource.into())?;
+    let inputs = read_inputs(input_path)?;
+    let (proof, report) = zkvm.prove(&inputs).with_context(|| "Failed to prove")?;
+    fs::write(&proof_path, proof)
+        .with_context(|| format!("Failed to write proof to {}", proof_path.display()))?;
+    fs::write(
+        &report_path,
+        bincode::serialize(&report).with_context(|| "Failed to serialize report")?,
+    )
+    .with_context(|| format!("Failed to write report to {}", report_path.display()))?;
+    Ok(())
+}
+
+fn execute(program_path: PathBuf, input_path: PathBuf, report_path: PathBuf) -> Result<(), Error> {
+    let zkvm = construct_zkvm(program_path, Default::default())?;
+    let inputs = read_inputs(input_path)?;
+    let report = zkvm.execute(&inputs).with_context(|| "Failed to execute")?;
+    fs::write(
+        &report_path,
+        bincode::serialize(&report).with_context(|| "Failed to serialize report")?,
+    )
+    .with_context(|| format!("Failed to write report to {}", report_path.display()))?;
+    Ok(())
+}
+
+fn verify(program_path: PathBuf, proof_path: PathBuf) -> Result<(), Error> {
+    let zkvm = construct_zkvm(program_path, Default::default())?;
+    let proof = fs::read(&proof_path)
+        .with_context(|| format!("Failed to read proof from {}", proof_path.display()))?;
+    zkvm.verify(&proof)
+        .with_context(|| "Failed to verify proof")?;
+    Ok(())
+}
+
+fn construct_zkvm(
+    program_path: PathBuf,
+    resource: zkvm_interface::ProverResourceType,
+) -> Result<impl zkVM, Error> {
+    let program = fs::read(&program_path)
+        .with_context(|| format!("Failed to read program at {}", program_path.display()))?;
+
+    #[cfg(feature = "jolt")]
+    {
+        let program =
+            bincode::deserialize(&program).with_context(|| "Failed to deserialize program")?;
+        ere_jolt::EreJolt::new(program, resource).with_context(|| "Failed to instantiate EreJolt")
+    }
+
+    #[cfg(feature = "nexus")]
+    {
+        let program =
+            bincode::deserialize(&program).with_context(|| "Failed to deserialize program")?;
+        Ok(ere_nexus::EreNexus::new(program, resource))
+    }
+
+    #[cfg(feature = "openvm")]
+    {
+        let program =
+            bincode::deserialize(&program).with_context(|| "Failed to deserialize program")?;
+        ere_openvm::EreOpenVM::new(program, resource)
+            .with_context(|| "Failed to instantiate EreOpenVM")
+    }
```

Comment:
Could we create an issue, re this: If we add a bunch of zkVMs, then adding them all into this function doesn't really scale